Improve connection handling

wiegandm opened the pull request:
This commit improves the way connections are handled by the
`handle_client_stream()` method.

Previously, data from the client socket was read until a timeout was
reached after two seconds, i.e. when there was no activity from the open
client socket for two seconds. Beyond reaching this timeout or
encountering an error, there was no way to exit the `while True:` loop
containing the `recv` and `read` calls.

This lead to problems when a client sent data and waited less than two
seconds before closing the connection. Since the timeout was never
reached, the `data.append()` method continued to append `bytes` to the
`data` array indefinitely.

This commit introduces a conditional to check if any additional data was
read and to `break` the loop if not, i.e. all data sent by the client
was consumed.

This has the added benefit of reducing the time the client has to wait
after it is done sending by the two seconds the server previously waited
before starting to process the data.

wiegandm committed:
Improve connection handling

This commit improves the way connections are handled by the
`handle_client_stream()` method.

Previously, data from the client socket was read until a timeout was
reached after two seconds, i.e. when there was no activity from the open
client socket for two seconds. Beyond reaching this timeout or
encountering an error, there was no way to exit the `while True:` loop
containing the `recv` and `read` calls.

This lead to problems when a client sent data and waited less than two
seconds before closing the connection. Since the timeout was never
reached, the `data.append()` method continued to append `bytes` to the
`data` array indefinitely.

This commit introduces a conditional to check if any additional data was
read and to `break` the loop if not, i.e. all data sent by the client
was consumed.

This has the added benefit of reducing the time the client has to wait
after it is done sending by the two seconds the server previously waited
before starting to process the data.

bjoernricks commented on ospd/ospd.py:
```diff
@@ -777,9 +777,13 @@ def handle_client_stream(self, stream, is_unix=False):
                 else:
-                    data.append(stream.read(1024))
+                    buf = stream.read(1024)
+                if len(buf):
```
I would change this into
```suggestion
                if not buf:
                    # remote has closed the connection
                    break

                data.append(buf)
```

wiegandm replied:
Indeed, I have updated the PR. Upon closer inspection I have also removed the superfluous conditional following it.

wiegandm committed:
Simplify buffer check

This commit further simplifies the check for received data by inverting
the conditional and taking into account that an empty array is `False`.

It also removes the useless conditional checking the length of the
`data` array. The `data` array cannot be zero-length at this point
because at least an empty `bytes` object will have been appended to it,
resulting in a non-zero length.

bjoernricks committed:
Merge branch 'master' into connection_handling
